### PR TITLE
Fix custom container log dir do not exist

### PR DESCRIPTION
**Pull request by xlgao-zju**
We will not create the log dir for now, if custom container log dir do not exist.

cc @feiskyer 
Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

### CODE REVIEW

File: daemon/pod/container.go [hunk 1]
```diff
@@ -922,6 +922,14 @@ func (c *Container) initLogger() {
 	if c.p.factory.logCfg.Type == jsonfilelog.Name {
 		if c.spec.LogPath != "" {
 			ctx.LogPath = c.spec.LogPath
+			logdir := filepath.Dir(ctx.LogPath)
+
+			if err := os.MkdirAll(logdir, os.FileMode(0755)); err != nil {
```
Comment (feiskyer) [inline daemon/pod/container.go]: dup with else block, it's better move the code after else block.

Comment (xlgao-zju) [inline daemon/pod/container.go]: ACK

**Comment by xlgao-zju** [comment]
@feiskyer updated

**Comment by feiskyer** [comment]
LGTM. Thanks.

**Pull request merged**